addons/mail/models/mail_thread.py

```diff
@@ -1344,7 +1344,7 @@ def _message_route_process(self, message, message_dict, routes):
         self = self.with_context(attachments_mime_plainxml=True) # import XML attachments as text
         # postpone setting message_dict.partner_ids after message_post, to avoid double notifications
         original_partner_ids = message_dict.pop('partner_ids', [])
-        thread_id = False
+        thread = None
```

Comment:
If the API is now to return a recordset, this should be an empty recordset rather than None.